tests/e2e/specs/BoundaryEventValidatiion.spec.js

```diff
@@ -0,0 +1,44 @@
+import { dragFromSourceToDest, getComponentsEmbeddedInShape, getElementAtPosition, setBoundaryEvent } from '../support/utils';
+import { nodeTypes } from '../support/constants';
+import uniqWith from 'lodash/uniqWith';
+import isEqual from 'lodash/isEqual';
+
+describe('Boundary event validation', function() {
+  it('should add boundary events to empty ports around boundary event target, and not allow adding any more', function() {
+    const taskPosition = { x: 250, y: 200 };
+    dragFromSourceToDest(nodeTypes.task, taskPosition);
+
+    const numberOfPortsAroundTask = 12;
+    for (let i = 0; i < numberOfPortsAroundTask; i++) {
+      setBoundaryEvent(nodeTypes.boundaryMessageEvent, taskPosition, nodeTypes.task);
```

Comment:
nodeTypes.task is the default if you want to shorten these.

Reply:
I'm not sure I understand. Why would I probably want `nodeTypes.subProcess`?
For ports, the same validation should apply regardless of the specific target task type or boundary event type. The decision to use the base task type and the message boundary event was arbitrary.

Reply:
This was before we merged in my ticket to add Message Events to all Task types. You can simplify it to:
`setBoundaryEvent(nodeTypes.boundaryMessageEvent, taskPosition);`